src/components/IKUpload/props.ts

```diff
@@ -0,0 +1,30 @@
+import PropTypes, { InferProps } from 'prop-types';
+import React, { LegacyRef } from 'react';
+
+const Props = {
+    fileName: PropTypes.string,
+    tags: PropTypes.arrayOf(PropTypes.string.isRequired),
+    useUniqueFileName: PropTypes.bool,
+    responseFields: PropTypes.array,
+    isPrivateFile: PropTypes.bool,
+    folder: PropTypes.string,
+    customCoordinates: PropTypes.string,
+    onError: PropTypes.func,
+    onSuccess: PropTypes.func,
+    onUploadStart: PropTypes.func,
+    onUploadProgress: PropTypes.func,
+    inputRef: PropTypes.object,
+    validateFile: PropTypes.func,
```

Comment:
It should have proper interface similar to JS SDK - https://github.com/imagekit-developer/imagekit-javascript/blob/master/src/interfaces/UploadOptions.ts

Reply:
React has a concept of type checking which is enforced at runtime and this is the one which we have used. Apart from this we have a call to `InferProps` which will automatically derive the type definitions for typescript (which are only compile time) from the code of react that is defining runtime types. This prevents code duplication.